Security Team

j01tz opened the pull request:
[Rendered link to RFC document](https://github.com/mimblewimble/grin-rfcs/blob/d6295f25e728599dc49fe848ba97a45bfd8d7315/text/0000-security-team.md)

This is a proposal to create a security team for Grin.

lehnberg commented:
@j01tz it was brought to my attention that as both security team and core team are intermixed in the document, `vetoes` that the security team members have might mistakenly be read as applying to core team decisions. Perhaps it could be made clearer that the vetoes only apply on security team decisions?

Also following gov meeting just now, would be great if you could drop "sub-team" for just "team" instead in the doc. :)

j01tz commented:
Following your comment, I'm debating language use in veto section
>Any member of the security team may veto an impactful **security-related** decision

or

>Any member of the security team may veto an impactful **security team** decision

The former gives the team the ability to broadly veto _any_ impactful security-related decision (though doesn't guarantee its awareness of these situations). The latter restricts it to "security team decisions" which would be narrowly interpreted as what is specified in the RFC document and what is explicitly delegated by core.

Depending on the approach we could do a better job in the RFC of spelling out what are explicitly security team decisions if we don't want to leave "security-related decisions" up to interpretation.

lehnberg commented:
I’d be inclined to go with the latter: **security team decision**

Practically speaking, the remits of the security team is contained within the security team itself. It cannot get veto powers on matters that are not handled by the team.

j01tz commented:
Thanks very much for the feedback.

* Changed the "subteam" language to reflect updates from the meeting
* Added language to clarify "security team decisions" for vetoes and deadlocks

lehnberg commented:
Looks good @j01tz, bar any objections from others I'm going to recommend that this RFC moves into **Final Comment Period** with a **disposition to merge** tomorrow during the governance meeting. :+1:

lehnberg commented:
Following discussion in today's Governance meeting, and in line with our governance process, this RFC can be considered being in **Final Comment Period** from today, with a disposition to **merge** in two weeks time, on February 25.

Please ensure any comments are made before then! ⏳

lehnberg commented on text/0000-security-team.md:
_Previous_ process

lehnberg commented on text/0000-security-team.md:
_Security Team Process_

lehnberg commented on text/0000-security-team.md:
If all membership is vetted and approved by the core team, do we really need to have "one well represented core developer" here as a requirement? Feels a bit redundant. If core doesn't feel the need to have a member on the team, that's even better, and not sure why it would need to be a requirement.

j01tz replied:
I agree, I removed this requirement :+1:

lehnberg commented on text/0000-security-team.md:
Similarly, this entire section might then be removed.

j01tz replied:
Likewise, removed this section as well.

lehnberg commented on text/0000-security-team.md:
This sentence reads a bit unclear to me, can it be rephrased?

j01tz replied:
Changed line to:
>In the event that the core team does not follow a security team recommendation, a reasonable explanation must be provided as to why the security recommendation was not followed

Does that make more sense?

lehnberg commented on text/0000-security-team.md:
Previous process

lehnberg commented on text/0000-security-team.md:
Security team process

lehnberg commented on text/0000-security-team.md:
How is this a drawback for the formation of a security team?

j01tz replied:
Clarified here, thanks :+1:

lehnberg commented on text/0000-security-team.md:
Another alternative that could be highlighted is to not have a dedicated team at all and instead just rely on an open community to handle all incidents as they come in. It might be wroth writing a couple of lines why this might be less desirable than a dedicated team.

j01tz replied:
Took a shot at exploring this alternative, let me know if I'm missing any important pros or cons.

lehnberg commented on text/0000-security-team.md:
This might be redundant if there's not one always appointed member requirement

j01tz replied:
Agreed and updated

j01tz commented:
Thanks for the feedback @lehnberg. I think I was able to address your comments. I also added a line for clarification in the decision making section.

yeastplume commented:
Looks good to go as is, thanks @j01tz for putting all of this together.

lehnberg commented:
Following discussion in yesterday's Governance meeting, and in line with our governance process, following two weeks of final comment period, this RFC can now be considered **Accepted**. 🎉
